### ISSUE TITLE
iOS Simulator hangs at loading

### ISSUE THREAD

**Issue by aurnik** [Microsoft/vscode-react-native#383]
I apologize ahead of time if I'm reporting this incorrectly, I'm trying to migrate a React Native project started on Nuclide into VS Code and this happened right off the bat.

## Actual Behavior
1. The packager starts and then the iOS simulator starts the app but just hangs at the Loading screen (doesn't even start the loading process showing any percentages)

## Expected Behavior
1. iOS Simulator starts and goes into app which should start loading normally.

## Software versions
- React-Native VS Code extension version: 0.2.5
- VSCode version: 1.8.1
- OS platform and version: OS X El Capitan 10.11.6
- NodeJS version: 6.0.0
- React Native version: 0.38.0

## Outputs (Include if relevant)
- Output of the Debug Console (View -> Toggle Debug Console):
```
[Warning] Couldn't import script at <http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false>. Debugging won't work: Try reloading the JS from inside the app, or Reconnect the VS Code debugger: Unexpected token *
```
- Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):
```
Getting exponent dependecy.
######### Checking react native is correctly setup. This may take a while... ##########
Checking if the correct react native is installed..
######### Starting Packager ##########
######### Executing command: react-native start --port 8081 ##########
Scanning 774 folders for symlinks in /Users/aurnik/Documents/HYDR8/node_modules (5ms)
 ┌────────────────────────────────────────────────────────────────────────────┐ 
 │  Running packager on port 8081.                                            │ 
 │                                                                            │ 
 │  Keep this packager running while developing on any JS projects. Feel      │ 
 │  free to close this tab and run your own packager instance if you          │ 
 │  prefer.                                                                   │ 
 │                                                                            │ 
 │  https://github.com/facebook/react-native                                  │ 
 │                                                                            │ 
 └────────────────────────────────────────────────────────────────────────────┘ 
Looking for JS files in
   /Users/aurnik/Documents/HYDR8 

[Hot Module Replacement] Server listening on /hot

React packager ready.

[1/18/2017, 10:35:04 PM] <START> Initializing Packager
[1/18/2017, 10:35:04 PM] <START> Building in-memory fs for JavaScript
[1/18/2017, 10:35:04 PM] <END>   Building in-memory fs for JavaScript (164ms)
[1/18/2017, 10:35:04 PM] <START> Building Haste Map
[1/18/2017, 10:35:04 PM] <END>   Building Haste Map (71ms)
[1/18/2017, 10:35:04 PM] <END>   Initializing Packager (305ms)
######### Packager started. ##########
[1/18/2017, 10:35:05 PM] <START> Requesting bundle bundle_url: /index.ios.bundle?platform=ios
[1/18/2017, 10:35:05 PM] <START> Transforming files
[1/18/2017, 10:35:05 PM] <END>   Transforming files (213ms)
######### The Bundle Cache was prewarmed. ##########
[1/18/2017, 10:35:05 PM] <END>   Requesting bundle bundle_url: /index.ios.bundle?platform=ios (300ms)
[1/18/2017, 10:35:15 PM] <START> Requesting bundle bundle_url: /index.ios.bundle?platform=ios&dev=true&minify=false
[1/18/2017, 10:35:15 PM] <START> Transforming files
[1/18/2017, 10:35:16 PM] <END>   Transforming files (332ms)
[1/18/2017, 10:35:16 PM] <START> Requesting bundle bundle_url: /index.ios.bundle?platform=ios&dev=true&minify=false
Starting custom debugger by executing: echo A debugger is not needed:  "/Users/aurnik/Documents/HYDR8"
[1/18/2017, 10:35:16 PM] <END>   Requesting bundle bundle_url: /index.ios.bundle?platform=ios&dev=true&minify=false (455ms)
[1/18/2017, 10:35:16 PM] <END>   Requesting bundle bundle_url: /index.ios.bundle?platform=ios&dev=true&minify=false (118ms)
[1/18/2017, 10:35:17 PM] <START> Requesting bundle bundle_url: /index.ios.bundle?platform=ios&dev=true&minify=false
[1/18/2017, 10:35:17 PM] <END>   Requesting bundle bundle_url: /index.ios.bundle?platform=ios&dev=true&minify=false (16ms)
[1/18/2017, 10:35:17 PM] <START> Requesting bundle bundle_url: /index.ios.map?platform=ios&dev=true&minify=false
[1/18/2017, 10:35:17 PM] <END>   Requesting bundle bundle_url: /index.ios.map?platform=ios&dev=true&minify=false (62ms)
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
[Extension Host] Error: Module version mismatch. Expected 50, got 48.
    at Error (native)
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:178:20)
    at Object.Module._extensions..node (module.js:583:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:178:20)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Function.o._load (/Users/aurnik/Downloads/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24306)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/aurnik/.vscode/extensions/vsmobile.vscode-react-native-0.2.5/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)A.logExtensionHostMessage @ extensionHost.ts:293A.onMessaage @ extensionHost.ts:232(anonymous function) @ extensionHost.ts:171emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98

[Extension Host] Error: Cannot find module './build/default/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Function.o._load (/Users/aurnik/Downloads/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24306)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/aurnik/.vscode/extensions/vsmobile.vscode-react-native-0.2.5/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Function.o._load (/Users/aurnik/Downloads/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24306)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/aurnik/.vscode/extensions/vsmobile.vscode-react-native-0.2.5/node_modules/bunyan/lib/bunyan.js:79:18)
    at Module._compile (module.js:556:32)A.logExtensionHostMessage @ extensionHost.ts:293A.onMessaage @ extensionHost.ts:232(anonymous function) @ extensionHost.ts:171emitTwo @ events.js:106emit @ events.js:191process.nextTick @ internal/child_process.js:744_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98

[Extension Host] Error: Cannot find module './build/Debug/DTraceProviderBindings'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Function.o._load (/Users/aurnik/Downloads/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24306)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/aurnik/.vscode/extensions/vsmobile.vscode-react-native-0.2.5/node_modules/dtrace-provider/dtrace-provider.js:17:23)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Function.o._load (/Users/aurnik/Downloads/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/node/extensionHostProcess.js:12:24306)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/aurnik/.vscode/extensions/vsmobile.vscode-react-native-0.2.5/node_modules/bunyan/lib/bunyan.js:79:18)
    at Module._compile (module.js:556:32)
```

**Comment by MSLaguana** [comment]
Does the same thing happen for you if you `react-native run-ios` on the command line?

**Comment by aurnik** [comment]
No that's the unusual thing, running it from the command line works completely fine.

**Comment by MSLaguana** [comment]
Can you please try launching to iOS with our extension, then when it fails go to the command line and run `curl 'http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false' > bundle.js`, and finally try running `node bundle.js` and see whether that throws an error.

The error above indicates that the compiled JS bundle has some kind of parse error in it, but I'm not sure why that would happen only for us and not when running via the command line. If you can reproduce the error in node it might point to the line, and if you could post the section in the bundle it complains about we might be able to work out what's going wrong.

**Comment by aurnik** [comment]
I figured it out! (Sort of)
I think it was an issue on my end, I removed the node_modules folder, reset the cache, reinstalled node, and then updated my packages and for some reason it works now. I still don't understand why the issue only appeared in VSCode and not the terminal, but it was a package issue rather than a vscode-react-native issue. Sorry about that!
Thanks for all your help though @MSLaguana
